content/webapp/pages/search/index.tsx

```diff
@@ -276,61 +368,85 @@ const NewSearchPage: NextPageWithLayout<NewProps> = ({
-                              {
-                                query: queryString,
-                                workType: [bucket.data.id],
-                              },
-                              `works_workType_${pathname}`
-                            )}
-                            passHref
-                          >
-                            {bucket.data.label} ({bucket.count})
-                          </NextLink>
-                        )
-                      )}
-                      <NextLink
-                        {...worksLink(
-                          {
-                            query: queryString,
-                          },
-                          `works_all_${pathname}`
-                        )}
-                        passHref
-                      >
-                        All Catalogue results{' '}
-                        {catalogueResults.works?.totalResults}
-                      </NextLink>
-                    </div>
+              {(!!(
+                catalogueResults.images?.totalResults &&
+                catalogueResults.images?.totalResults > 0
+              ) ||
+                !!(
+                  catalogueResults.works?.totalResults &&
+                  catalogueResults.works?.totalResults > 0
+                )) && (
+                <CatalogueResultsInner>
+                  {catalogueResults.works &&
+                    catalogueResults.works.totalResults > 0 && (
+                      <>
+                        <CatalogueSectionTitle sectionName="Catalogue results" />
+                        <CatalogueLinks>
+                          {catalogueResults.works.workTypeBuckets?.map(
+                            (bucket, i) => (
+                              <NextLink
+                                key={i}
+                                {...worksLink(
+                                  {
+                                    query: queryString,
+                                    workType: [bucket.data.id],
+                                  },
+                                  `works_workType_${pathname}`
+                                )}
+                                passHref
+                                legacyBehavior
+                              >
+                                <WorksLink>
+                                  {bucket.data.label} ({bucket.count})
+                                </WorksLink>
+                              </NextLink>
+                            )
+                          )}
+                        </CatalogueLinks>
+                        <NextLink
+                          {...worksLink(
+                            {
+                              query: queryString,
+                            },
+                            `works_all_${pathname}`
+                          )}
+                          passHref
+                          legacyBehavior
+                        >
```

Comment:
These links don't work for me locally, there's no href in the element
![Uploading Screenshot 2025-02-03 at 12.10.18.png…]()

Reply:
hmm, I'm struggling to replicate this

Reply:
I pulled the latest, running with `run-concurrently` and both of first and subsequent loads these links don't work for me 🤔

Reply:
sorry, I'm looking at the wrong links!

Reply:
tbf I might've commented the wrong one 😅

Reply:
I read up about legacyBehavior but didn't think to check the links!

Reply:
That'd be the last thing for me, otherwise good to go and very nice to see!